src/me/truemb/rentit/utils/UtilMethodes.java

```diff
@@ -508,6 +511,7 @@ public void sendList(Player p, RentTypes type, int site) {
 	}
 
 	public List<Inventory> getShopChestInventories(int shopId) {
+		// TODO pending
```

Comment:
This method is still used by the `ShopCacheFileManager` in order to set a backup of the shop. I couldn't figure out exactly what is this backup so I didn't want to change it. However, the backup will not get the items stored in the new type of chests if we keep it like this. We should include a new method in the ChestUtils that can support the backup needs.

Reply:
The backup is for **/shop rollback**. Users can go to their old shop and get the items back. Even if the rent contract run out.
The rollback contains all the items of the chests in the shop area.

Reply:
The ```getShopChestInventories``` goes through every block in the shop and gets the items in chests.

```
Inventory chestInv = chest.getBlockInventory(); //Only get the Block Inventory, so it doesnt get counts twice on double chests.
chestInventories.add(chestInv); //Adds the inventory to the List.
```

Reply:
How do they get it back exactly? It's pretty simple to get the list of all items using the new ChestUtils but if we need to put those items inside specific chests or stuff like it we might need to implement some extra logic to handle it.

Reply:
Just some Inventories, that are the content for the rollbackinventory. Its nothing special. Players can just get items of the inventory, but cant put any in.

https://github.com/TrueMB/RentIt/blob/8612df9ad5132d2bb19db27e81818aeed93b60f6/src/me/truemb/rentit/data/RollbackInventoryData.java#L11

Reply:
I changed the implementation from `setShopBackup` to pull from the new list of items returned by the SupportedChest wrapper. `getAllItems` might return more than one full inventory for paginated chests like the ones of AdvancedChests but all the rest in the backup code should remain the same.